content/PHPDataObjects/DataObjects.q.md

```diff
@@ -0,0 +1,1145 @@
+
+- It prints `Successfully connected to the database.` and `Successfully inserted the values into the table.`.
+
+
+/// type=SS, answer=[5]
+
+Which statement correctly describes the query `INSERT INTO students (firstname, lastname, birthdate, gender) VALUES ('Samantha', 'Danes', '1999-10-12', 'Female')`?
+
+- It modifies the `students` table.
+
+- It deletes the data in the `students` table
+
+- It replaces the values in the `students` table.
+
+- It removes the values `John` and `Smith` in the `students` table.
+
+- It adds a new row that contains the values `Samantha`,`Danes`, `1999-10-12`, and `Female` in the `students` table.
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+  $host = 'localhost';
+  $db = 'LibraryDB';
+  $port = '5432';
+  $username = 'postgres';
+  $password = 'Admin01';
+
+  $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+  try {
+      $conn = new PDO($dsn);
+      if ($conn) {
+          echo "Successfully connected to the database. ";
+      }
+  } catch (Exception $e) {
+      echo "Unable to establish a connection."; 
+  }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once ("connection.php");
+
+    $stmt = $conn->query("SELECT * FROM students");
+    foreach ($stmt as $row) {
+        echo $row['student_id'].$row['firstname'].$row['lastname'].$row['birthdate'].$row['gender'].' ';
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `2SamanthaDanes1999-10-12Female`.
+
+- It prints `Successfully connected to the database.` and `1JohnSmith1999-02-10Male 2SamanthaDanes1999-10-12Female`.
+
+
+/// type=SS, answer=[5]
+
+In `student.php`, remove `.$row['lastname']` on line 6. Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `2Samantha1999-10-12Female 1John1999-02-10Male`.
+
+- It prints `Successfully connected to the database.` and `1John1999-02-10Male 2Samantha1999-10-12Female`.
+
+
+/// type=SS, answer=[2]
+
+In `student.php`. remove `as $row` in the `foreach` statement on line 5. Execute the program. What is the error message?
+
+- Undefined variable: `row` on line 5
+
+- syntax error, unexpected `)` on line 5
+
+- syntax error, unexpected `foreach` on line 5
+
+- Invalid argument supplied for `foreach()` on line 5 
+
+- syntax error, unexpected `}`, expecting the end of file on line 5
+
+
+/// type=SS, answer=[5]
+
+Which statement correctly describes the error?
+
+- The `students` table is miswritten as `student` on line 4.
+
+- In `student.php`, there is no opening bracket `{` on line 5.
```

Comment:
Incorrect term use here... In PHP we did not use `opening` and also in PHP it is not bracket but curly brace. Please check. Thanks.

Please rephrase... 

opening brace `{` --->  open curly brace `{`